software/base/src/test/java/brooklyn/entity/basic/SoftwareProcessEntityTest.java

```diff
@@ -568,4 +706,85 @@ protected String getInstallLabelExtraSalt() {
             return (String)getEntity().getConfigRaw(ConfigKeys.newStringConfigKey("salt"), true).or((String)null);
         }
     }
+
+    public static class ReleaseLatchLocation extends SimulatedLocation {
+        private static final long serialVersionUID = 1L;
+        
+        private AtomicBoolean lock = new AtomicBoolean();
+        private boolean isBlocked;
+
+        public void unblock() {
+            synchronized (lock) {
```

Comment:
Given it's for a test, I don't really care but...

I much prefer using higher-level synchronization utils from java.util.concurrent whenever possible, rather than synchronize.
An object.wait() can wake up prematurely, which is why one should always wrap it in a while block rather than an if block.